rules/do-you-know-when-to-mockup-your-api/rule.md

```diff
@@ -1,27 +1,132 @@
 related: 
 - do-you-know-how-to-brand-your-api-portal
 - do-you-know-the-best-tools-to-manage-apis
 redirects:
-- comments-do-you-know-when-to-mockup-your-apis
+- do-you-know-when-to-mockup-your-apis
 
 ---
 
-Designing your API up front can be very useful where you have a significant amount of backend development to complete before there is sufficient functionality for front end development to easily commence. 
+Designing your API up front can be very useful where you have a significant amount of backend development to complete before there is sufficient functionality for front end development to easily commence.
+
 <!--endintro-->
-Rather than waiting for the backend to be complete, the APIs can be agreed up front. Tools like **Azure API Management** then allow you to mock up the APIs with sample data, allowing the front end development to commence much sooner.  
- This can result in valuable feedback on the API and backend functionality much earlier in your development lifecycle. The key benefit is being able to present the frontend to stakeholders for feedback really early.  
 
+Rather than waiting for the backend to be complete, the APIs can be agreed on up front. Tools like Azure API Management then allow you to mock up the APIs with sample data, allowing the front end development to commence much sooner.
+
+This can result in valuable feedback on the API and backend functionality much earlier in your development lifecycle. The key benefit is being able to present the frontend to stakeholders for feedback really early.
 
 Watch Adam Cogan and William Liebenberg discuss the benefits.
 
-`youtube: https://youtu.be/nKb6kj_lXXM`
+`youtube: https://youtu.be/rNyOALskc_U`
+
+## Steps to create and test a mocked API endpoint
+
+To complete the following steps, you need to create an instance of Azure API Management in your subscription.
+
+> TIP: If you have Azure Credits, the Developer (NO SLA) edition is the cheapest option to test most of the features offered by API Management.
+
+### Step 1 - Create Blank API
+
+1. Select **Blank API**
+
+![](step1-create-blank-api.png)
+Figure: Create a blank API
+
+## Step 2 - Describe your API
+
+![](step2-describe-your-api.png)
+Figure: Describe your API
+
+1. Click **Create**
+
+## Step 3 - Add an Operation
+
+1. Select your API from the API list
+2. Click **Add operation**
+3. Specify the Display name, Name and URL of your operation 
+
+![](step3-add-operation.png)
+Figure: Adding a new Operation
+
+## Step 4 - Add response
+
+1. Click **Add response**
+2. Select the appropriate response code (e.g 200 OK)
+
+![](step4-add-response.png)
+Figure: Add a response
+
+## Step 5 - Add representation
+
+1. Specify the **Content Type** as `application/json`
+2. Specify the **Sample** as `{ "data": "fake" }`
+3. Click **Save**
+  
+![](step5-add-represtntation.png)
+Figure: Add a representation for your response with fake data
+
+![](step5-operation-done.png)
+Figure: Operation added successfully
```

Comment:
The figure should go inside the square brackets, the text won't show up on GitHub previews but will when deployed.

Reply:
done